lib/webpacker/manifest.rb

```diff
@@ -17,31 +17,86 @@ def file_path
       Webpacker::Configuration.manifest_path
     end
 
-    def lookup(name)
+    # Converts the "name" (aka the pack or bundle name) to to the full path for use in a browser.
+    def asset_source(name)
+      output_path_or_url = Webpacker::Configuration.output_path_or_url
+      mapped_name = lookup(name)
+      "#{output_path_or_url}/#{mapped_name}"
+    end
+
+    # Converts the "name" (aka the pack or bundle name) to the possibly hashed name per a manifest.
+    #
+    # If Configuration.compile? then compilation is invoked the file is missing.
+    #
+    # Options
+    # throw_if_missing: default is true. If false, then nill is returned if the file is missing.
+    def lookup(name, throw_if_missing: true)
+      instance.confirm_manifest_exists
+
       if Webpacker::Configuration.compile?
-        compile_and_find!(name)
+        compile_and_find!(name, throw_if_missing: throw_if_missing)
       else
-        find!(name)
+        # Since load checks a `mtime` on the manifest for a non-production env before loading,
+        # we should always call this before a call to `find!` since one may be using
+        # `webpack -w` and a file may have been added to the manifest since Rails first started.
+        load
```

Comment:
Forgive the drive-by comment, but framework code should not be directly checking the environment: we have a config subsystem for that. People often have a `production`-alike `staging` environment, for example.

(More locally-relevant, this should likely use the file watcher to avoid the stat [where possible] even in development.)